flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlCreateTable.java

```diff
@@ -186,9 +195,24 @@ public void validate() throws SqlValidateException {
             }
         }
 
+        if (query != null && tableLike != null) {
+            throw new SqlValidateException(
+                    pos,
+                    "CREATE TABLE AS SELECT and CREATE TABLE LIKE syntax cannot be used together.");
+        }
+
         if (tableLike != null) {
             tableLike.validate();
         }
+
+        if (query != null
+                && (columnList.size() > 0
+                        || constraints.size() > 0
+                        || partitionKeyList.size() > 0)) {
+            throw new SqlValidateException(
+                    pos,
+                    "CREATE TABLE AS SELECT syntax does not yet support Column/Partition/Constraints settings.");
```

Comment:
may be some more clear message like
`CREATE TABLE AS SELECT syntax does not yet support to specific Column/Partition/Constraints`